### PR TITLE
#331 refactor regarding viewBinding, update kotlin and gradle plugin

### PR DESCRIPTION
**Pull request by Marchuck**
Hello guys, I refactored fragments, removed all findViewById / synthetic  and introduced ViewBinding according to [this feature request](https://github.com/android/uamp/issues/331)

I'm happy I could contribute to **uamp**!

### CODE REVIEW

**Comment by Marchuck** [comment]
@googlebot I signed it!

File: gradle.properties [hunk 1]
```diff
@@ -25,7 +25,7 @@ android.enableJetifier=true
 android.useAndroidX=true
 org.gradle.jvmargs=-Xmx1536m
 # Required by Robolectric.
-android.enableUnitTestBinaryResources=true
+#android.enableUnitTestBinaryResources=true
```
Comment (nic0lette) [inline gradle.properties:28]: QQ: Do the tests still run fine given this change? I feel like I needed this at one point?

File: app/src/main/res/layout/fragment_mediaitem_list.xml [hunk 2]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (nic0lette) [inline app/src/main/res/layout/fragment_mediaitem_list.xml:1]: nit: Please make sure the `<!--` is on the following line (the formatter loves to move it there x.x)

**Comment by nic0lette** [comment]
Thanks for the contribution! Just the one question and nit, but otherwise looks good!

File: gradle.properties (hunk 1 above)
Comment (Marchuck) [reply gradle.properties:28]: according to [docs](http://robolectric.org/getting-started/), this flag is no longer needed

File: app/src/main/res/layout/fragment_mediaitem_list.xml (hunk 2 above)
Comment (Marchuck) [reply app/src/main/res/layout/fragment_mediaitem_list.xml]: done

File: app/build.gradle [hunk 3]
```diff
@@ -38,9 +38,15 @@ android {
             useSupportLibrary true
         }
     }
+
+    viewBinding {
+        enabled = true
+    }
+
```
Comment (dturner) [inline app/build.gradle:45]: The convention now is to replace `viewBinding {} ` and `dataBinding {}` with a `buildFeatures` block: 

```
android
    buildFeatures{
        viewBinding true
        dataBinding true
    }
}
```

Comment (Marchuck) [reply app/build.gradle]: thanks, done

**Comment by dturner** [comment]
Thanks for submitting this PR - your contributions are very much appreciated!

**Pull request merged**